packages/url/src/clean-for-slug.js

```diff
@@ -24,7 +24,7 @@ export function cleanForSlug( string ) {
 		return '';
 	}
 	return (
-		removeAccents( string )
+		removeAccents( string.replace( /&amp;/g, '' ).replace( /&lt;/g, '' ) )
```

Comment:
The `cleanForSlug` function is intended to be a mirror of [`sanitize_title()`](https://developer.wordpress.org/reference/functions/sanitize_title/), which doesn't account for HTML entities. This is also confirmed by the `cleanForSlug` documentation, which specifically claims it [doesn't touch any HTML entities](https://github.com/WordPress/gutenberg/blob/a3360a8f228a6eea778c130c327c0374f56d6352/packages/url/src/clean-for-slug.js#L15-L16). With that in mind, perhaps there is a better place to address this change?

Reply:
Yes, I wondered why it wasn't a mirror – there is a core default filter added to `sanitize_title()` which runs the `$title` through [`sanitize_title_with_dashes()`](https://developer.wordpress.org/reference/functions/sanitize_title_with_dashes/#source) and the expectation when doing PHP development is that `sanitize_title( 'the dog & the cat' )` will return `the-dog-the-cat`. 

It's that core filter which handles the heavy lifting of sanitization and removal of the html entities among other things, this is the reason the string returned by `cleanForSlug()` doesn't match the actual slug returned by `sanitize_title()`.

People might be using this and changing the returned value could horribly mess their urls up – what about keeping back compat and introducing a new function `sanitizeForSlug()` which mirrors `sanitize_title()` with the default filter added (ie add in all the sanitization done by `sanitize_title_with_dashes()`) and then using `sanitizeForSlug()` when displaying the slug. 

Does that approach sound ok?

Reply:
> there is a core default filter added to sanitize_title() which runs the $title through [sanitize_title_with_dashes()](https://developer.wordpress.org/reference/functions/sanitize_title_with_dashes/#source) and the expectation when doing PHP development is that sanitize_title( 'the dog & the cat' ) will return the-dog-the-cat.

This makes sense, thanks for pointing it out 👍 

> People might be using this and changing the returned value could horribly mess their urls up – what about keeping back compat and introducing a new function sanitizeForSlug() which mirrors sanitize_title() with the default filter added (ie add in all the sanitization done by sanitize_title_with_dashes()) and then using sanitizeForSlug() when displaying the slug.

> Does that approach sound ok?

A separate utility function sounds good to me 👍 We'll definitely want to keep `cleanForSlug` as it is, since it intentionally doesn't handle entities.